feature/migration/qrcode/src/main/kotlin/app/k9mail/feature/migration/qrcode/QrCodePayloadValidator.kt

```diff
@@ -0,0 +1,187 @@
+package app.k9mail.feature.migration.qrcode
+
+import app.k9mail.core.common.mail.EmailAddress
+import app.k9mail.core.common.mail.EmailAddressParserException
+import app.k9mail.core.common.mail.toUserEmailAddress
+import app.k9mail.core.common.net.Hostname
+import app.k9mail.core.common.net.Port
+import app.k9mail.core.common.net.toHostname
+import app.k9mail.core.common.net.toPort
+import timber.log.Timber
+
+@Suppress("TooManyFunctions")
```

Comment:
This is a good indicator that this is doing too much, e.g. validation the input data and transformation of `QrCodeData` into `AccountData`.